quantecon/discrete_var.py

```diff
@@ -0,0 +1,159 @@
+        -   N is an m x 1 vector containing the number of grid points in the discretization of each element of x_t, By default, N = 10
+        -   T is the the length of the simulated time series. By default, T = 1_000_000
+        -   Tburn is the number of burn-in draws from the simulated series. By default, Tburn = 100_000
+        -   UB is an m x 1 vector indicating the upper bound value of the grid. The grid will be simmetrical around 0. By default, 
+            UB = sqrt(10)*std(x_t(i))
+        
+        Outputs:
+        -   Pi is an squared-matrix containing the transition probability matrix of the discretized state. By default, the code removes 
+            the states that are never visited. 
+        -   S is a matrix where the element (i,j) of S is the discretized value of the j-th element of x_t in state i. Reducing S to its
+            unique values yields the grid values.
+        -   Xvec is a matrix of size m by T containing the simulated time series of the m discretized states.    
+        
+        Ancillary Functions:
+        -   Mom function: computes the VAR(1)'s unconditional variance.
+        
+        # Example:
+
+        -   This example discretizes  the sthocastic process used to calibrate the economic model included in ``Downward Nominal Wage Rigidity, 
+            Currency Pegs, and Involuntary Unemployment'' by Stephanie Schmitt-Grohé and Martín Uribe, Journal of Political Economy 
+            124, October 2016, 1466-1514. 
+                
+                A     = np.array([[0.7901, -1.3570], [-0.0104, 0.8638]])
+                omega = np.array([[0.0012346, -0.0000776], [-0.0000776, 0.0000401]])
+                N = np.array([21, 11])
+                Pii, Xvec, S = tpm(A, omega, N, T=1_000_000, Tburn = 100_000)
+
+        
+    """
+        
+    m = len(A)
+    r = len(omega)
+    
+    if np.isnan(np.sum(N)):
+        N = 10 * np.ones(m, dtype = int)
+    
+    Sigg = mom(np.eye(m), A, omega)
+    sigg = np.sqrt(np.diagonal(Sigg))  # Unconditional  standard deviation  of AR process
+    
+    
+    if np.isnan(np.sum(UB)):
+        UB = np.sqrt(10) * sigg
+    
+    V = []
+    for i in range(m):
+        b = np.linspace(-UB[i], UB[i], N[i])
+        V.append(b)
+
+    n = N.prod()     # Total number of possible values of the discretized state
+    S = np.zeros([n, m])
+    
+    for i in range(m):
+        if i == 0:
+            temp    = np.ravel(V[i])
+            S[:, i] = np.ravel(np.tile(temp,[np.prod(N[i+1:]), 1] ))
+        else:
+            temp = np.sort(np.ravel(np.tile(V[i],[np.prod(N[0:i]), 1])))
+            S[:, i] = np.ravel(np.tile(temp,[np.prod(N[i+1:]), 1] ))
+
+    Pi = np.zeros([n, n])
+    x0 = np.zeros([m, 1])
+    xx = np.zeros([n, m])
+    d = np.sum((S-xx)**2, 1)
+    ind_i = np.argmin(d)
+    Xvec = np.zeros([m, T])
+    mean = [0] * m
+    
+    for t in range(T+Tburn):
+        drw = multivariate_normal(mean, omega).reshape(1,2)
```

Comment:
@crondonm , does this line assume that `m = 2` ?  I'm guessing `2` should be `m` in my edit.

Reply:
Hi @jstac,

m should be equal to the number of states you are discretizing. In the example, m = 2.

Reply:
Thanks @crondonm .  In the specified example that's true but for the code to work for arbitrary examples we need to replace 2 by m.  I've taken care of this.